### ISSUE TITLE
Specify directory stucture per path instead of per project in config-user.yml

### ISSUE THREAD

**Issue by bouweandela** [ESMValGroup/ESMValCore#129]
At the moment all drs need to be the same per project, while it would make much more sense to be able to specify the drs per rootpath.

To make this more clear, now we have this:
```yaml
rootpath:
  # Rootpath to CMIP5 data
  CMIP5:
    - /badc/cmip5/data/cmip5/output1
    - ~/some_other_path

# Directory structure for input data: [default]/BADC/DKRZ/ETHZ/etc
# See config-developer.yml for definitions.
drs:
  CMIP5: BADC
```
while I think this would be much more useful:
```yaml
# Directory structure for input data in the format
# input_data:
#   project_a:
#     /path/to/dir_1: drs_1
#     /path/to/dir_2: drs_2
#   project_b:
#     /path/to/dir_3: drs_2
# See config-developer.yml for valid drs definitions per project.
input_data:
  CMIP5:
    /badc/cmip5/data/cmip5/output1: BADC
    ~/some_other_path: default
```

**Comment by bouweandela** [comment]
Related to #181

**Comment by bouweandela** [comment]
I'll try to fix this for v2.8.

**Comment by bouweandela** [comment]
Bumping to the next milestone as there is no consensus on the implementation, see discussion in the linked pull request.
